src/models/__tests__/message_view.test.ts

```diff
@@ -66,6 +66,51 @@ const containerMock = ({
   }
 } as unknown) as Container;
 
+const aNoticeNumber = "177777777777777777";
+
+describe("message_view", () => {
+  it("GIVEN a valid message_view obejct WHEN the object is decode THEN the decode succeed", async () => {
+    const result = MessageView.decode(aMessageView);
+    expect(E.isRight(result)).toBeTruthy();
+  });
+
+  it("GIVEN a valid message_view obejct with payment WHEN the object is decode THEN the decode succeed", async () => {
+    const messageViewWithPayment = {
+      ...aMessageView,
+      components: {
+        ...aMessageView.components,
+        payment: { has: true, notice_number: aNoticeNumber }
+      }
+    };
+    const result = MessageView.decode(messageViewWithPayment);
+    expect(E.isRight(result)).toBeTruthy();
+  });
+
+  it("GIVEN a message_view obejct with a missing notice_number payment WHEN the object is decode THEN the decode fails", async () => {
+    const messageViewWithPayment = {
+      ...aMessageView,
+      components: {
+        ...aMessageView.components,
+        payment: { has: true }
+      }
+    };
+    const result = MessageView.decode(messageViewWithPayment);
+    expect(E.isLeft(result)).toBeTruthy();
+  });
+
+  it("GIVEN a message_view obejct with a notice_number-only payment WHEN the object is decode THEN the decode fails", async () => {
+    const messageViewWithPayment = {
+      ...aMessageView,
+      components: {
+        ...aMessageView.components,
+        payment: { has: false, notice_number: aNoticeNumber }
+      }
+    };
+    const result = MessageView.decode(messageViewWithPayment);
+    expect(E.isRight(result)).toBeTruthy();
+  });
```

Comment:
Title says that we are expecting this decode to fail

Reply:
fixed, thank you!